lib/ex_doc/retriever.ex

```diff
@@ -568,4 +567,40 @@ defmodule ExDoc.Retriever do
   # Cut off the body of an opaque type while leaving it on a normal type.
   defp process_type_ast({:::, _, [d|_]}, :opaque), do: d
   defp process_type_ast(ast, _), do: ast
+
+  # Since Elixir 1.7.0-dev, functions from Kernel.Typespec related with the
+  # runtime aspects, such as reading from .beam files, were moved to Code.Typespec
+  if Version.compare(System.version(), "1.7.0-dev") == :lt do
```

Comment:
I think I would prefer using `if Code.ensure_loaded?(Code.Typespec) do` as the condition instead.